### PR TITLE
upload dialog for simple dataset creation

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Import GeoJSON](https://trello.com/c/PEQWEkHp/48-import-geojson)

File: projects/wave-core-new/src/lib/datasets/dataset.service.ts [hunk 1]
```diff
@@ -5,13 +5,14 @@ import {DataSet} from './dataset.model';
...
-import {CreateDataSetDict, DataSetIdDict, UploadResponseDict} from '../backend/backend.model';
+import {AutoCreateDataSetDict, CreateDataSetDict, DataSetIdDict, DatasetIdResponseDict, UploadResponseDict} from '../backend/backend.model';
+import {RandomColorService} from '../util/services/random-color.service';
 
 @Injectable({
     providedIn: 'root',
 })
 export class DataSetService {
-    constructor(private backend: BackendService, private userService: UserService) {}
+    constructor(private backend: BackendService, private userService: UserService, private randomColorService: RandomColorService) {}
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/datasets/dataset.service.ts:15]: Why do you add it and not use it?
If you touch it, you could set the others to protected.

File: projects/wave-core-new/src/lib/datasets/upload/upload.component.scss [hunk 2]
```diff
@@ -18,5 +18,5 @@ mat-spinner {
 }
 
 textarea {
-    width: 100%;
+    width: 99%;
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/datasets/upload/upload.component.scss:21]: What is this about?

File: projects/wave-core-new/src/lib/project/project.service.ts [hunk 3]
```diff
@@ -712,6 +717,79 @@ export class ProjectService {
         return ProjectService.subscribeAndProvide(result);
     }
 
+    addDataSetToMap(dataset: DataSet): Observable<void> {
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/project/project.service.ts:720]: You should put it into your dataset service. I don't see why it should be in the project service.

File: projects/wave-core-new/src/lib/tabs/tabs.component.scss [hunk 4]
```diff
@@ -37,7 +37,6 @@
 ::ng-deep .mat-tab-header {
     height: 3rem;
 
-    margin-left: 3rem;
     @media (max-width: 600px) and (orientation: portrait) {
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/tabs/tabs.component.scss:40]: Why did you remove this?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/wave-core-new/src/lib/datasets/upload/upload.component.scss (hunk 2 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datasets/upload/upload.component.scss:21]: If I leave it at 100% I get a scrollbar, because the textarea is now inside a tabcomponent. Not sure how to properly fix this.

File: projects/wave-core-new/src/lib/project/project.service.ts (hunk 3 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/project/project.service.ts:720]: I put it there first but I was afraid of getting a circular dependency between Project and DatasetService. And adding something to the map might as well be done in ProjectService I think

File: projects/wave-core-new/src/lib/tabs/tabs.component.scss (hunk 4 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/tabs/tabs.component.scss:40]: because it made the tabs in the upload dialog look weird and I wasn't able to overwrite this setting. As I don't see where else it is used I removed it.

File: projects/wave-core-new/src/lib/datasets/dataset.service.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datasets/dataset.service.ts:15]: I first added the addToMap method there, if it stays in ProjectService (cf other comment) I will remove it.

File: projects/wave-core-new/src/lib/tabs/tabs.component.scss (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/tabs/tabs.component.scss:40]: :host ::ng-deep …

**Comment by ChristianBeilschmidt** [comment]
* Frameset -> no border
* inputs ->  100% width
* input background: all grey
*

File: projects/wave-core-new/src/lib/project/project.service.ts (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/project/project.service.ts:720]: you don't have a dependency from project to datasets

File: projects/wave-core-new/src/lib/datasets/dataset.service.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/datasets/dataset.service.ts]: 6b05d3ccd0295b16ae979a4cca1f1a468a5d2489

File: projects/wave-core-new/src/lib/project/project.service.ts (hunk 3 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/project/project.service.ts]: 6b05d3ccd0295b16ae979a4cca1f1a468a5d2489

File: projects/wave-core-new/src/lib/tabs/tabs.component.scss (hunk 4 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/tabs/tabs.component.scss]: 45ca50031043e21f611db6f3ae861c14f7f756ab

**Comment by michaelmattig** [comment]
> 
> 
>     * Frameset -> no border
> 
>     * inputs ->  100% width
> 
>     * input background: all grey
> 
>     * disabled leads to warnings in console

45ca50031043e21f611db6f3ae861c14f7f756ab
